pkg/profiles/profiles.go

```diff
@@ -18,6 +20,45 @@ type profileTemplateConfig struct {
 	ClusterZone           string
 }
 
+// DefaultRetryBudget is used for routes which do not specify one.
+var DefaultRetryBudget = pb.RetryBudget{
+	MinRetriesPerSecond: 10,
+	RetryRatio:          0.2,
+	Ttl: &duration.Duration{
+		Seconds: 10,
+	},
+}
+
+// ToServiceProfile returns a Proxy API DestinationProfile, given a
+// ServiceProfile.
+func ToServiceProfile(profile *sp.ServiceProfileSpec) (*pb.DestinationProfile, error) {
+	routes := make([]*pb.Route, 0)
+	for _, route := range profile.Routes {
+		pbRoute, err := ToRoute(route)
+		if err != nil {
+			return nil, err
+		}
+		routes = append(routes, pbRoute)
+	}
+	budget := DefaultRetryBudget
+	if profile.RetryBudget != nil {
```

Comment:
Sorry if this is addressed elsewhere and I couldn't find it: would this be an appropriate place to validate the budget arguments? Would an error here be more visible to a user?

The proxy will already disregard invalid budgets and log at WARN level, which should be visible, but only if the user is looking at the proxy logs...

Reply:
Good point.  Probably a good idea to validate here too.  Unfortunately, the suffers from the same problem that validation errors will only be visible if you're looking at the logs.  

We may be able to some limited validation with CRD schema validation too...

Reply:
Is it possible for `linkerd inject` or other related command to do the validation then?

Reply:
It'd be a great addition to the `inject` checks.